lib/hanami/configuration.rb

```diff
@@ -33,7 +36,12 @@ class Configuration
     attr_reader :environments
     private :environments
 
-    def initialize(env:)
+    attr_reader :application_name
+
+    def initialize(application_name:, env:)
+      @namespace = application_name.split(MODULE_DELIMITER)[0..-2].join(MODULE_DELIMITER)
```

Comment:
Thinking it through further, it actually feels kind of odd to have the determination of `namespace` as part of the `Configuration` class. This isn't something that's going to be configurable at all, and it should really be the responsibility of the application class itself to tell the rest of the system about its namespace. Do you reckon there's a way we can keep it back in `Application`?

It's kind of the same thing for the `application_name`, too, really. It's a value that's _useful_ for parts of the configuration to know, as we've seen, but it's not really the _job_ of the `Configuration` to determine the name in the first place (unless, of course, we want to make it a truly configurable setting...)

Reply:
@timriley I refactored this under the need of accessing the `application_name`, without thinking about the responsibilities between application and configuration.

To avoid cyclic dependencies between the two, I moved the namespace logic from application to configuration.

I consider namespace in the configuration, a **private utility for us**, NOT something that we need to configure.

Of course, if you have a suggestion, we can apply it. 🙂